### PR TITLE
Adds the "To Number" circuit component

### PR DESCRIPTION
**Pull request by TheSmallBlue**
## About The Pull Request
While you can calmly turn a number into a string by just linking the number output and the string input, you cannot do the same with turning a string into a number. This will change with these super duper cash money 40 fucking lines of code hell yeah

## Why It's Good For The Game
Allows for taking stuff from a string, for example a message, and into other components like index list.
Example: 
https://youtu.be/IW03YapF0pk

## Changelog
:cl:
expansion: Added the To Number circuit component.
/:cl:

### CODE REVIEW

**Comment by Mothblocks** [comment]
Shorten the name to just "To Number", I think the string is implied.

**Comment by TheSmallBlue** [comment]
I'm kinda scared that people will think it takes in any value if I name it "To Number", since "To String" does exactly that, but sure.

**Comment by Mothblocks** [comment]
They'll learn once they try 😄

File: code/modules/wiremod/components/string/tonumber.dm [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+/obj/item/circuit_component/tonumber/input_received(datum/port/input/port)
+	. = ..()
+	if(.)
+		return
+
+	var/input_value = input_port.input_value
+
+	if(isnull(input_value))
+		return
```
Comment (Watermelon914) [inline code/modules/wiremod/components/string/tonumber.dm:37]: set the output to null

Comment (Watermelon914) [reply code/modules/wiremod/components/string/tonumber.dm:37]: in fact, just don't have this check at all. It'll be set to null anyways when it sets the output to output_value

File: code/modules/wiremod/components/string/tonumber.dm [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+/obj/item/circuit_component/tonumber/input_received(datum/port/input/port)
+	. = ..()
+	if(.)
+		return
+
+	var/input_value = input_port.input_value
+
+	var/output_value = text2num(input_value)
+
+	output.set_output(output_value)
```
Comment (Mothblocks) [inline code/modules/wiremod/components/string/tonumber.dm:38]:
```suggestion
	output.set_output(text2num(input_port.input_value))
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Pull request merged**